scylla/tests/ccm-integration/cluster.rs

```diff
@@ -0,0 +1,832 @@
+                        )?;
+                }
+                _ => {
+                    return Err(Error::from(err).context(format!(
+                        "failed to create root directory {}",
+                        opts.config_dir()
+                    )));
+                }
+            },
+        }
+
+        let lcmd = LoggedCmd::new(opts.ccm_log_file())
+            .await
+            .with_context(|| format!("failed to create command logger {}", opts.ccm_log_file()))?;
+
+        let mut cluster = Cluster {
+            destroyed: false,
+            nodes: NodeList::new(),
+            logged_cmd: Arc::new(lcmd),
+            opts: opts.clone(),
+        };
+
+        for datacenter_id in 0..opts.nodes.len() {
+            for _ in 0..opts.nodes[datacenter_id] {
+                cluster.append_node(Some((datacenter_id + 1) as u16)).await;
+            }
+        }
+
+        Ok(cluster)
+    }
+
+    pub(crate) async fn init(&mut self) -> Result<(), Error> {
+        let cluster_dir = PathBuf::from(self.opts.cluster_dir());
+        let config_dir = self.opts.config_dir();
+        if cluster_dir.exists() {
+            tokio::fs::remove_dir_all(&cluster_dir)
+                .await
+                .with_context(|| {
+                    format!(
+                        "failed to remove config dir {}",
+                        cluster_dir.to_str().unwrap()
+                    )
+                })?;
+        }
+        let mut args: Vec<String> = vec![
+            "create".to_string(),
+            self.opts.name.clone(),
+            "-v".to_string(),
+            self.opts.version.clone(),
+            "-i".to_string(),
+            self.opts.ip_prefix.to_string(),
+            "--config-dir".to_string(),
+            config_dir.clone(),
+        ];
+        if self.opts.db_type == DBType::SCYLLA {
+            args.push("--scylla".to_string());
+        }
+        self.logged_cmd
+            .run_command("ccm", &args, RunOptions::None)
+            .await?;
+        let nodes_str = self
+            .opts
+            .nodes
+            .iter()
+            .map(|node| node.to_string())
+            .collect::<Vec<String>>()
+            .join(":");
+
+        let args = vec![
+            "populate".to_string(),
+            "-i".to_string(),
+            self.opts.ip_prefix.to_string(),
+            "-n".to_string(),
+            nodes_str,
+            "--config-dir".to_string(),
+            config_dir.clone(),
+        ];
+        self.logged_cmd
+            .run_command("ccm", &args, RunOptions::None)
+            .await?;
+        Ok(())
+    }
+
+    fn get_ccm_env(&self) -> HashMap<String, String> {
+        let mut env: HashMap<String, String> = HashMap::new();
+        env.insert(
+            "SCYLLA_EXT_OPTS".to_string(),
+            format!("--smp={} --memory={}M", self.opts.smp, self.opts.memory),
+        );
+        env
+    }
```

Comment:
I am unable to imagine why would someone think it is a good idea to pass some options throught env instead normal arguments.....

Reply:
)))